docs/docs/20-usage/90-advanced-usage.md

```diff
@@ -141,17 +141,65 @@ The snippet below shows how a step can communicate with the docker daemon via a
 If your aim ist to build/publish OCI images, consider using the [Docker Buildx Plugin](https://woodpecker-ci.org/plugins/Docker%20Buildx) instead.
 :::
 
-The most important bit ist to allow the dind service to write/read certificates on a mounted volume that is also shared with the step that will contact the daemon (it needs the service's CA to do so successfully).
+First we need to define a servie running a docker with the `dind` tag. This service must run in privileged mode:
 
-The `DOCKER_*` environment variables in the step are generic settings that should work with the docker client used by your framework of choice (e.g. [TestContainers](https://testcontainers.com/), [Spring Boot Docker Compose](https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-docker-compose) or similar):
+```yaml
+services:
+  - name: docker
+    image: docker:27.4-dind
+    privileged: true
+    ports:
+      - 2376
+```
+
+Next we need to set up TLS communication between the `dind` service and the step that wants to communicate with the docker daemon (since Unauthenticated TCP connections have been deprecated [as of docker v26](https://github.com/docker/cli/blob/v27.4.0/docs/deprecated.md#unauthenticated-tcp-connections) and will ve removed in release v28).
+
+We can achieve this by letting the daemon generate TLS certificates for us and share them with the client via a volume mount in the agent (`/opt/woodpeckerci/dind-certs` in the example below).
+
+```diff
+services:
+  - name: docker
+    image: docker:27.4-dind
+    privileged: true
++    environment:
++      DOCKER_TLS_CERTDIR: /dind-certs
++    volumes:
++      - /opt/woodpeckerci/dind-certs:/dind-certs
+     ports:
+       - 2376
+```
+In the step that needs access to the daemon we need to:
+
+1. Set the `DOCKER_*` environment variables shown below, setting up the connection with the daemon. These are standardized environment variables that should work with the docker client used by your framework of choice (e.g. [TestContainers](https://testcontainers.com/), [Spring Boot Docker Compose](https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-docker-compose) or similar).
+2. Mount the volume where the daemon has created the certificates (`/opt/woodpeckerci/dind-certs`)
+
+In this example we test the connection with the vanilla docker client:
+
+```diff
+steps:
+  - name: test
+    image: docker:27.4-cli
++    environment:
++      DOCKER_HOST: "tcp://docker:2376"
++      DOCKER_CERT_PATH: "/dind-certs/client"27.4-cli
++      DOCKER_TLS_VERIFY: "1"
++    volumes:
++      - /opt/woodpeckerci/dind-certs:/dind-certs
+    commands:
+      - docker version
+```
+
+This step should output version information of the client and the server if everything has been set correctly.
+
+Complete example:
 
 ```yaml
 steps:
   - name: test
-    image: docker:latest
```

Comment:
I meant to just use a comment here for the major version hint and keep `latest`. Otherwise the docs don't age well.